crates/subspace-farmer/src/cluster/controller/farms.rs

```diff
@@ -213,15 +241,19 @@ pub async fn maintain_farms(
                         );
                     }
 
-                    None
+                    FarmAddRemoveResult::Remove { farm_index }
                 }));
 
                 match result {
                     Ok(()) => {
                         info!(%farm_index, "Farm exited successfully");
                     }
                     Err(error) => {
-                        error!(%farm_index, %error, "Farm exited with error");
+                        // Notify for cleanup the farmer if it exited with an error
+                        error!(%farm_index, %error, "Farm exited with error, notify for cleanup");
+                        known_farmers
+                            .get_mut_farmer_by_farm_index(farm_index)
+                            .map(KnownFarmer::notify_cleanup);
```

Comment:
Please don't violate abstractions calling cleanups externally.

If farm failed, it will be removed. If removed farm is the last one, then the whole farmer need to go down. There is no need to do manual cleanups like this, also there is no need to crash the whole farmer if just one farm has crashed (it is happening sometimes with bad drives, etc., and it is important for farmer overall to continue farming in the meantime).